pkg/apis/discovery/validation/validation.go

```diff
@@ -131,8 +131,8 @@ func validatePorts(endpointPorts []discovery.EndpointPort, fldPath *field.Path)
 		idxPath := fldPath.Index(i)
 
 		if len(*endpointPort.Name) > 0 {
-			for _, msg := range validation.IsValidPortName(*endpointPort.Name) {
-				allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), endpointPort.Name, msg))
+			for _, msg := range validation.IsDNS1123Label(*endpointPort.Name) {
```

Comment:
`allErrs = append(allErrs, apivalidation.ValidateDNS1123Label(...`

Reply:
How would I do that while maintaining the other information that I'm adding within the loop? `IsDNS1123Label` only returns a slice of strings as error messages, and in this loop I'm adding the `field.Invalid` error type, the path of this invalid port name, and the value that is invalid, all in addition to the actual message returned from `IsDNS1123Label`. This is the only way I've seen it done elsewhere, but admittedly this pattern is only really used in the core validation package other than this.

Reply:
ValidateDNS1123Label constructs the Invalid field errors, using the field path you pass in

Reply:
See use in validating endpoint ports here:

https://github.com/kubernetes/kubernetes/blob/b6bf0d6d8e2f19cf6b7840a7443e0e892da7697d/pkg/apis/core/validation/validation.go#L5447

Reply:
That's actually a helper function within the core validation package that calls the function I'm using `validation.IsDNS1123Label` (https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L118). It felt strange to require the core validation package here for that helper func, and overkill to rewrite that function here. Should I require the core validation package here to use that helper function?